opentracing/span.py

```diff
@@ -209,17 +211,23 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         """Ends context manager and calls finish() on the :class:`Span`.
 
-        If exception has occurred during execution, it is automatically added
-        as a tag to the :class:`Span`.
+        If exception has occurred during execution, it is automatically logged
+        and added as a tag. :attr:`~operation.ext.tags.ERROR` will also be set
+        to `True`.
         """
-        if exc_type:
-            self.log_kv({
-                'python.exception.type': exc_type,
-                'python.exception.val': exc_val,
-                'python.exception.tb': exc_tb,
-                })
+        self._on_error(exc_type, exc_val, exc_tb)
         self.finish()
 
+    def _on_error(self, exc_type, exc_val, exc_tb):
+        if not exc_type:
+            return
+
+        self.set_tag(tags.ERROR, True)
+        self.log_kv({
+            logs.EVENT: tags.ERROR,
+            logs.ERROR_OBJECT: exc_type,
```

Comment:
I don't have a good understanding of Python exceptions, but this seems odd. Wouldn't this make more sense?

* error.kind = exc_type
* error.object = exc_val (??)
* stack = exc_tb

Reply:
Oh, I should win a prize for being so careless here, as it should have been `exc_val` - and also, I was trying to use the advised behavior from https://github.com/opentracing/specification/blob/master/semantic_conventions.md so we it should be enough to have the error object.

Reply:
can the tracer implementation still extract the stack trace if you only pass `exc_val`?  What's the reason that Python splits those three?

Reply:
Oh, good point - so, no, it's not possible in Python 2 at least. My guess was to go and append it as another log field, but I also see we don't have any `stack` field, as the Specification declares (this also happens in Java).

In any case, I think I will go ahead and add such traceback anyway. Let me know - the log field name can be added later on ;)